public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx

```diff
@@ -34,6 +35,15 @@ type Props = {
   onChange: (value: DataSourceSettings<ElasticsearchOptions>) => void;
 };
 export const ElasticDetails = ({ value, onChange }: Props) => {
+  const currentVersion = esVersions.find((version) => version.value === value.jsonData.esVersion);
+  const customOption =
+    (!currentVersion &&
+      valid(value.jsonData.esVersion) && {
+        label: value.jsonData.esVersion,
+        value: value.jsonData.esVersion,
+      }) ||
+    undefined;
+
```

Comment:
i did not run this yet, but cannot this be done with a ternary, like this? (i find that a little easier to understand in this case)


```suggestion
  const customOption =
    !currentVersion && valid(value.jsonData.esVersion)
      ? {
          label: value.jsonData.esVersion,
          value: value.jsonData.esVersion,
        }
      : undefined;
```

Reply:
yup, makes sense! 👍